flink-batch-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCInputFormat.java

```diff
@@ -157,14 +203,25 @@ public boolean reachedEnd() throws IOException {
 	 * @throws java.io.IOException
 	 */
 	@Override
-	public OUT nextRecord(OUT tuple) throws IOException {
+	public Row nextRecord(Row row) throws IOException {
 		try {
-			resultSet.next();
-			if (columnTypes == null) {
-				extractTypes(tuple);
+			hasNext = resultSet.next();
+			if(!hasNext) {
+				return null;
+			}
+			try{
+				//This throws a NPE when the TypeInfo is not passed to the InputFormat,
+				//i.e. KryoSerializer used to generate the passed row
+				row.productArity();
+			}catch(NullPointerException npe) {
```

Comment:
Ok thanks! I saw that there's a discussion about a formatter for IntelliJ...it would be great to have such a formatter also for Eclipse :)